### PR TITLE
Add TIFF writer that supports floating point images

### PR DESCRIPTION
**Pull request by csparker247**

### CODE REVIEW

File: apps/src/ExtractENVI.cpp [hunk 1]
```diff
@@ -103,13 +104,15 @@ int main(int argc, char** argv)
         // Select file extension
         fs::path out = outputDir / (envi.getWavelength(band) + ".png");
 
-        if (m.depth() == CV_32F) {
-            out.replace_extension("hdr");
-        } else if (m.depth() == CV_16U) {
+        if (m.depth() == CV_16U || m.depth() == CV_32F || m.depth() == CV_64F) {
```
Comment (aconway01) [inline apps/src/ExtractENVI.cpp:107]: Would clang-format not separate this into multiple lines?

File: envi-tools/src/TIFFIO.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    lt::TIFFSetField(out, TIFFTAG_SAMPLESPERPIXEL, channels);
+    lt::TIFFSetField(out, TIFFTAG_ROWSPERSTRIP, rowsPerStrip);
+
+    // Row buffer. OpenCV documentation mentions that TIFFWriteScanline
+    // modifies its read buffer, so we can't use the cv::Mat directly
+    auto bufferSize = static_cast<size_t>(lt::TIFFScanlineSize(out));
+    std::vector<char> buffer(bufferSize + 32);
+
+    // For each row
+    for (unsigned row = 0; row < height; row++) {
```
Comment (aconway01) [inline envi-tools/src/TIFFIO.cpp:103]: auto row, unless the assignment would default to an int

Comment (csparker247) [reply envi-tools/src/TIFFIO.cpp:103]: It does default to an int. It's determined by the lhs of the assignment, in this case the 0.

**Review by aconway01 — APPROVED** [review]

File: apps/src/ExtractENVI.cpp (hunk 1 above)
Comment (csparker247) [reply apps/src/ExtractENVI.cpp:107]: It's right at 80 characters. *fist pump*

**Pull request merged**